src/transports/ipc/bipc.c

```diff
@@ -168,6 +168,11 @@ static void nn_bipc_destroy (struct nn_epbase *self)
 static void nn_bipc_shutdown (struct nn_fsm *self, int src, int type,
     void *srcptr)
 {
+#if !defined NN_HAVE_WINDOWS
```

Comment:
Its stil a reasonable question — the presence of the #define doesn’t mean
that IPC is actually implemented.  Indeed, I’ve some evidence that IPC
doesn’t work quite the same everywhere — it can, for example, depend on the
filesystem code supporting it.  For example, I believe it doesn’t work in
all directories on QNX (or so I’ve heard).  As another example, it doesn’t
work on FAT filesystems where you need to have support for pipe special
files.

That said, I think we can leave this question up to the platform code — if
the code *compiles* then that’s all we can do.  If it fails at runtime,
then the user is going to have to work around it.

For the case of Windows, we don’t use AF_UNIX at all, and I’m happy to have
a per-platform feature test for it — Windows is such a special snowflake
here (with its named pipes) that it deserves special handling.

 - Garrett


On Tue, Apr 26, 2016 at 8:04 AM, Franklin Mathieu <notifications@github.com>
wrote:

> In src/transports/ipc/bipc.c
> <https://github.com/nanomsg/nanomsg/pull/678#discussion_r61102523>:
>
> > @@ -168,6 +168,11 @@ static void nn_bipc_destroy (struct nn_epbase *self)
> >  static void nn_bipc_shutdown (struct nn_fsm *self, int src, int type,
> >      void *srcptr)
> >  {
> > +#if !defined NN_HAVE_WINDOWS
>
> Actually, nevermind: AF_UNIX on windows is defined in winsock2.h, so we
> can feature test with sys/socket.h.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/nanomsg/nanomsg/pull/678/files/45c6a91b373c82f2b0235de2d9fabb362522e6e6#r61102523>
>